### PR TITLE
Added option to set a backup time stamp

### PR DESCRIPTION
**Pull request by ravil-mobile**

### CODE REVIEW

File: backends/Base.h [hunk 1]
```diff
@@ -308,7 +316,10 @@ class Base
 		struct stat statBuffer;
 		if (m_rank == 0 && stat(file.c_str(), &statBuffer) == 0) {
 			logWarning() << file << "already exists. Creating backup.";
-			rename(file.c_str(), (file + ".bak_" + utils::TimeUtils::timeAsString("%F_%T", time(0L))).c_str());
+			if (!m_backupTimeStamp.has_value()) {
```
Comment (Thomas-Ulrich) [inline backends/Base.h:319]: this should in theory never happen (for seissol), as m_backupTimeStamp is initialized in src/main.cpp?

**Review by Thomas-Ulrich — APPROVED** [review]

File: backends/Base.h (hunk 1 above)
Comment (ravil-mobile) [reply backends/Base.h:319]: > this should in theory never happen (for seissol), as m_backupTimeStamp is initialized in src/main.cpp?

In the case of SeisSol, I totally agree. `XdmfWriter` can be used for other projects or sub-project where people do not care about having a unique timestamp for the entire application

Comment (Thomas-Ulrich) [reply backends/Base.h:319]: ok

**Review by Thomas-Ulrich — APPROVED** [review]

**Comment by ravil-mobile** [comment]
@Thomas-Ulrich , can you merge to the master b/c I don't have rights

**Comment by Thomas-Ulrich** [comment]
I don't have the rights either. @krenzland ?

**Pull request merged**